pkg/api/security/cert/cert_getter_dca.go

```diff
@@ -0,0 +1,136 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package cert
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net"
+	"net/url"
+
+	configModel "github.com/DataDog/datadog-agent/pkg/config/model"
+	configutils "github.com/DataDog/datadog-agent/pkg/config/utils"
+	"github.com/DataDog/datadog-agent/pkg/util/flavor"
+)
+
+// clusterCAData holds the cluster CA configuration and certificate data
+type clusterCAData struct {
+	enableTLSVerification bool
+	caCert                *x509.Certificate
+	caPrivKey             any
+}
+
+// readClusterCAConfig reads cluster CA configuration and files from disk once
+// Returns nil if no cluster CA is configured
+func readClusterCAConfig(config configModel.Reader) (*clusterCAData, error) {
+	enableTLSVerification := config.GetBool("cluster_trust_chain.enable_tls_verification")
+	clusterCAPath := config.GetString("cluster_trust_chain.ca_cert_file_path")
+	clusterCAKeyPath := config.GetString("cluster_trust_chain.ca_key_file_path")
+
+	// If no cluster CA path is configured, return nil (not an error)
+	if clusterCAPath == "" || clusterCAKeyPath == "" {
+		return &clusterCAData{
+			enableTLSVerification: enableTLSVerification,
+		}, nil
+	}
+
+	// Read cluster CA certificate and private key from disk
+	caCert, caPrivKey, err := ReadClusterCA(clusterCAPath, clusterCAKeyPath)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read cluster CA cert and key: %w", err)
+	}
+
+	return &clusterCAData{
+		enableTLSVerification: enableTLSVerification,
+		caCert:                caCert,
+		caPrivKey:             caPrivKey,
+	}, nil
+}
+
+// buildClusterClientTLSConfig creates the TLS configuration for cluster client communication
+// using pre-read cluster CA data
+func (c *clusterCAData) buildClusterClientTLSConfig() (*tls.Config, error) {
+	// Default to insecure configuration
```

Comment:
Nit: I would place the early return condition first 

```
if !c.enableTLSVerification {
    return &tls.Config{
		InsecureSkipVerify: true,
	}
}

...
```

That way is easier to read 😄